centreon/www/front_src/src/ApiTokens/TokenListing/Actions/Activate/Activate.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from 'react';
+
+import { useTranslation } from 'react-i18next';
+import { useQueryClient } from '@tanstack/react-query';
+
+import { Switch, Tooltip } from '@mui/material';
+
+import { Method, useMutationQuery } from '@centreon/ui';
+
+import { patchTokenEndpoint } from '../../../api/endpoints';
+import {
+  labelActive,
+  labelActiveOrRevoked,
+  labelRevoked
+} from '../../../translatedLabels';
+import { Row } from '../../models';
+
+import useActivateStyles from './Activate.styles';
+
+const Activate = ({ row }: Row): React.JSX.Element => {
+  const queryClient = useQueryClient();
+  const { t } = useTranslation();
+  const { classes } = useActivateStyles();
+
+  const [isRevoked, setIsRevoked] = useState<boolean>(row?.isRevoked);
+
+  useEffect(() => {
+    if (row?.isRevoked !== isRevoked) {
+      setIsRevoked(row?.isRevoked);
+    }
+  }, [row?.isRevoked]);
+
+  const { mutateAsync } = useMutationQuery({
+    getEndpoint: () =>
+      patchTokenEndpoint({ tokenName: row?.name, userId: row?.user.id }),
+    method: Method.PATCH,
+    onSuccess: () => queryClient.invalidateQueries({ queryKey: ['listTokens'] })
+  });
+
+  const onClick = (e: React.BaseSyntheticEvent): void => {
+    const value = e.target.checked;
+    setIsRevoked(value);
+
+    mutateAsync({
+      payload: { is_revoked: !value }
+    });
+  };
```

Comment:
You may need to handle the case where the request fails. In such a scenario, it's important to revert back to the previous state.

Reply:
Done